core/nodepool/config/templates/stack-template.json

```diff
@@ -101,7 +101,7 @@
               {{end}}
             ],
             "SubnetId": {{$workerSubnet.Ref}},
-            "UserData": {{ $.UserDataWorker.Parts.instance.Base64 true | checkSizeLessThan 16384 | quote }}
+            "UserData": {{ $.UserDataWorker.Parts.instance.Base64UserDataRef }}
```

Comment:
I've introduced `Base64UserDataRef` to inject `{"Ref":"AWS::StackName"}` into nodes by modifying the user-provided cloud-config(bash script)